app/components/admin/basic_link_post_component.rb

```diff
@@ -0,0 +1,18 @@
+module Admin
+  class BasicLinkPostComponent < ApplicationComponent
+    private attr_reader :title, :path, :name, :val
+
+    def initialize(title, path, name:, val:)
+      @title = title
+      @path = path
+      @val = val
+      @name = name
+    end
+
+    def call
+      form_with url: path do |form|
+        form.hidden_field(name, value: val) + form.submit(title, class: "hover:underline")
+      end
```

Comment:
I think we can get by with `button_to`. Looking at the documentation again I'm noticing we can pass `params` as HTML options.

https://apidock.com/rails/v5.2.3/ActionView/Helpers/UrlHelper/button_to

I think something like this might work and be a bit cleaner.

```ruby
button_to(title, path, {class: "hover:underline", params: {name => val}})
```

Reply:
yea, it works!!